readme/ImplementingRewrites.scalatex

```diff
@@ -0,0 +1,190 @@
+      the @lnk("Scalameta documentation", "http://scalameta.org/tutorial/#Scalahost").
+
+    @sect{Token}
+      A token is for example an identifier @code{println}, a delimiter @code{[} @code{)},
+      or a whitespace character like space or newline.
+      In the context of Scalafix, a @code{Token} means the data structure
+      @code{scala.meta.Token}.
+      See @lnk("Scalameta tutorial", "http://scalameta.org/tutorial/#Tokens")
+      for more details.
+      See @lnk("Wikipedia", "https://en.wikipedia.org/wiki/Lexical_analysis#Token")
+      for a more general definition.
+
+    @sect{Tokens}
+      @code{Tokens} is a list of @sect.ref{Token}.
+      See @lnk("Scalameta tutorial", "http://scalameta.org/tutorial/#Tokens")
+
+    @sect{Tree}
+      A @code{Tree} is a parsed abstract syntax tree.
+      In the context of Scalafix, a @code{Tree} means the data structure
+      @code{scala.meta.Tree}.
+      See @lnk("Scalameta tutorial", "http://scalameta.org/tutorial/#Trees")
+      for more details.
+      See @lnk("Wikipedia", "https://en.wikipedia.org/wiki/Abstract_syntax_tree")
+      for a more general definition.
+
+    @sect{Syntactic}
+      A @sect.ref{Rewrite} is "syntactic" when it does not require information
+      from type-checking such as resolved names (@code{println} => @code{scala.Predef.println}),
+      types or terms, or inferred implicit arguments.
+      A syntactic rewrite can use @sect.ref{Tokens} and @sect.ref{Tree}, but
+      not @sect.ref{Mirror}.
+
+    @sect{Semantic}
+      A @sect.ref{Rewrite} is "semantic" if it requires information from the compiler
+      such as types, symbols and reported compiler messages.
+      A semantic rewrite can use a @sect.ref{Mirror}.
+
+    @sect{Mirror}
+       A mirror is a Scalameta concept that encapsulates a compilation context, providing
+       capabilities to perform semantic operations for @sect.ref{Semantic} rewrites.
+       To learn more about mirrors and its associated concepts (Symbol,
+       Denotation, ...), see the
+       @lnk("Scalameta tutorial", "http://scalameta.org/tutorial/#Mirror").
+
+  @scalatex.Testkit()
+
+  @sect{Sharing your rewrite}
+    @p
+      You have implemented a rewrite, you have tests, it works,
+      and now you want to share it with the world. Congrats!
+      There are several ways to share a rewrite if the rewrite is contained in
+      a single file and uses no external dependencies,
+      @ul
+        @li
+          If you used @sect.ref{scalacenter/scalafix.g8} to build your project,
+          push your rewrite to github and tell users to run
+          @code{scalafix github:org/$reponame/$version}.
+        @li
+          otherwise, tell users to use the @sect.ref{http:} protocol,
+          @code{scalafix --rewrites https://gist....} where the url
+          points to the plaintext contents of your rewrite.
+
+      If your rewrite uses a custom library @code{foo}, then it's a bit tricky
```

Comment:
Good catch! I expanded on this part but decided to move it into the issue. Removed.